core/includes/backup/backup.mysql.inc

```diff
@@ -0,0 +1,583 @@
+   * That makes it easier to import without loading the whole file into memory.
+   * The files are a little harder to read, but human-readability is not a
+   * priority.
+   *
+   * @return int|boolean
+   *   If successful, the number of lines written. FALSE if the backup failed.
+   */
+  private function backupToFileInternal(BackupFile $file) {
+    $lines = 0;
+    $settings = $this->settings;
+    $exclude = !empty($settings['exclude_tables']) ? $settings['exclude_tables'] : array();
+    $nodata = !empty($settings['nodata_tables']) ? $settings['nodata_tables'] : array();
+    if ($file->open(TRUE)) {
+      $file->write($this->getSqlFileHeader());
+      $all_tables = $this->getTables();
+      $all_views = $this->getViews();
+
+      foreach ($all_tables as $table) {
+        if ($this->timeoutCheck()) {
+          return FALSE;
+        }
+        if ($table['name'] && !isset($exclude[$table['name']])) {
+          $file->write($this->getTableStructureSql($table));
+          $lines++;
+          if (!in_array($table['name'], $nodata)) {
+            $lines += $this->dumpTableDataSqlToFile($file, $table);
+          }
+        }
+      }
+      foreach ($all_views as $view) {
+        if ($this->timeoutCheck()) {
+          return FALSE;
+        }
+        if ($view['name'] && !isset($exclude[$view['name']])) {
+          $file->write($this->getViewCreateSql($view));
+        }
+
+      }
+      $file->write($this->getSqlFileFooter());
+      $file->close();
+      return $lines;
+    }
+    else {
+      return FALSE;
+    }
+  }
+
+  /**
+   * Backup the databases to a file using the mysqldump command.
+   */
+  private function backupToFileMysqldump(BackupFile $file) {
+    $nodata_tables = array();
+    $all_tables = $this->getTables();
+    $settings = $this->settings;
+
+    $command = 'mysqldump --result-file=%file --opt -Q --host=%host --port=%port --user=%user --password=%pass %name';
+    $args = array(
+      '%file' => $file->filepath(),
+      '%host' => $this->settings['database_host'],
+      '%port' => $this->settings['database_port'],
+      '%user' => $this->settings['database_user'],
+      '%pass' => $this->settings['database_pass'],
+      '%name' => $this->settings['database_name'],
+    );
+
+    // Ignore the excluded and no-data tables.
+    $db = $this->settings['database_name'];
+    if (!empty($settings['exclude_tables'])) {
+      foreach ((array) $settings['exclude_tables'] as $table) {
+        if (isset($all_tables[$table])) {
+          $command .= ' --ignore-table=' . $db . '.' . $table;
+        }
+      }
+    }
+    if (!empty($settings['nodata_tables'])) {
+      foreach ((array) $settings['nodata_tables'] as $table) {
+        if (isset($all_tables[$table])) {
+          $nodata_tables[] = $table;
+          $command .= ' --ignore-table=' . $db . '.' . $table;
+        }
+      }
+    }
+    // @todo: This is not yet supported.
+    $success = backup_migrate_exec($command, $args);
```

Comment:
I'm pretty sure, this backup_migrate_exec() is an oversight. :wink: 

There are more backup_migrate... functions below.

Reply:
I haven't port this bit yet since I'm considering if we should include `mysqldump` functionality at all. It's not on by default in B&M and it's probably not available on a good number of servers. Also `mysqldump` is no longer compatible with `mariadb` import. See https://ddev.com/blog/mariadb-dump-breaking-change/

It seems like we might be better suited if we narrowed our support to just the PHP-generated dumps and left more specific tooling to contrib or external tools.

Reply:
I went ahead and remove the `mysqldump` capability and use of `exec()` for the initial version of this feature.